addon/components/bs-modal-simple.hbs

```diff
@@ -1,24 +1,24 @@
 <BsModal
-  @open={{this.open}}
-  @fade={{this.fade}}
-  @backdrop={{this.backdrop}}
-  @keyboard={{this.keyboard}}
-  @position={{this.position}}
-  @scrollable={{this.scrollable}}
-  @size={{this.size}}
-  @backdropClose={{this.backdropClose}}
-  @renderInPlace={{this.renderInPlace}}
-  @transitionDuration={{this.transitionDuration}}
-  @backdropTransitionDuration={{this.backdropTransitionDuration}}
-  @onSubmit={{this.onSubmit}}
-  @onHide={{this.onHide}}
-  @onHidden={{this.onHidden}}
-  @onShow={{this.onShow}}
-  @onShown={{this.onShown}}
+  @open={{@open}}
+  @fade={{@fade}}
+  @backdrop={{@backdrop}}
+  @keyboard={{@keyboard}}
+  @position={{@position}}
+  @scrollable={{@scrollable}}
+  @size={{@size}}
+  @backdropClose={{@backdropClose}}
+  @renderInPlace={{@renderInPlace}}
+  @transitionDuration={{@transitionDuration}}
+  @backdropTransitionDuration={{@backdropTransitionDuration}}
+  @onSubmit={{@onSubmit}}
+  @onHide={{@onHide}}
+  @onHidden={{@onHidden}}
+  @onShow={{@onShow}}
+  @onShown={{@onShown}}
```

Comment:
Tests are failing due to the forwarding of arguments here. They do pass locally when `useDefaultValueIfUndefined`´ is set to true, as this is the same use case covered in #1248.

We could make it pass by wrapping every forwarding argument in a `{{bs-default}}`, requiring us to redefine all defaults here, which is pretty annoying. Also this brings up another issue: the `bs-default` helper actually does not follow the same semantics as the `@defaultValue` and `@arg` decorators, in that it will *always*  apply the default value if the argument is undefined. And there is no way around it, as it cannot distinguish between an argument not being used and and argument being passed with `undefined`. The only workaround - if we want to not risk any kind of breaking changes - would be to *not* use template-only components here and define defaults using `@arg`. 😟

This makes me think again if the approach in #1248 is really the best way forward, balancing the user's interests and ours as maintainers. Or if we shouldn't consider making the `useDefaultValueIfUndefined` behavior the default and only one. I think we still have a strong standpoint to not have to bump the major version, as eventually breaking code can be considered as relying on unintentional and undocumented behavior (e.g. I don't think there is anywhere in the API docs a reference to `undefined` as a legal value). Also I think there is a good chance that more or less no user is actually affected by this. So maybe we are making our lives way to hard here for no real (user's) benefit?

@jelhan thoughts?

Reply:
> This makes me think again if the approach in #1248 is really the best way forward, balancing the user's interests and ours as maintainers. Or if we shouldn't consider making the `useDefaultValueIfUndefined` behavior the default and only one.

I fully agree. I only introduced the `useDefaultValueIfUndefined` optional feature to unblock myself. There are strong use cases for it and I haven't come across any use case for the legacy behavior yet.

> I think we still have a strong standpoint to not have to bump the major version, as eventually breaking code can be considered as relying on unintentional and undocumented behavior (e.g. I don't think there is anywhere in the API docs a reference to `undefined` as a legal value). Also I think there is a good chance that more or less no user is actually affected by this. So maybe we are making our lives way to hard here for no real (user's) benefit?

I fully agree. As an alternative I would be also fine to have another major release. If it only includes these change and maybe drops `@triggerElement="parent-view"` upgrade should be easy for all applications. But we wouldn't compromise our strong SemVer guarantee.

Reply:
So, seems we have agreement here then. I assume you are still on vacation, so I will create another PR to remove the optional feature again (in favor of making it the default). Sorry for the additional work this has caused! 😔

Reply:
Actually I'm back from vacation since today. But have a huge list of open topics - as always after vacation. 🙄 So would be awesome if you could take care of this one.

Reply:
I rebased this on top of master which now includes #1283, so that should hopefully pass tests.